src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,105 @@
 package com.zipcodewilmington.looplabs;
 
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
-}
+    //constructor
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+        super(intArray);
+    }
+
+    Integer[] anIntArray = Arrays.copyOf(this.array, this.array.length);
+
+    @Override
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+
+        int noDuplicates = 0;
+        Integer[] duplicates = new Integer[0];
+        int duplicatesIndex = 0;
+        for (int i = 0; i < anIntArray.length; i++) {
+            if (contains(duplicates, anIntArray[i])) {
+                continue;
+            } else if (getNumberOfOccurrences(anIntArray, anIntArray[i]) >= maxNumberOfDuplications) {
+                duplicates = Arrays.copyOf(duplicates, duplicates.length + 1);
+                duplicates[duplicatesIndex] = array[i];
+                duplicatesIndex++;
+            }
+        }
+        Integer[] checking = Arrays.copyOf(anIntArray, anIntArray.length);
+        for (Integer k : duplicates) {
```

Comment:
I would just name `k` `duplicate` so that you could read this "for each duplicate in duplicates"

Reply:
Will update! Thanks for the feedback!